packages/layout/src/Box/Box.tsx

```diff
@@ -0,0 +1,89 @@
+import type t from "@telegraph/tokens";
+import clsx from "clsx";
+import React from "react";
+
+import { BOX_PROPS, VARIANT } from "./Box.constants";
+import { deriveSpacing } from "./Box.helpers";
+
+type BoxProp = keyof typeof BOX_PROPS;
+type BoxProps = React.HTMLAttributes<HTMLDivElement> & {
+  [key in BoxProp]?: `${keyof typeof t.tokens.spacing}`;
+} & {
+  // More variants wil beed added here once
+  // they are designed
+  variant?: "ghost";
+};
+type BoxRef = HTMLDivElement;
+
+const Box = React.forwardRef<BoxRef, BoxProps>(
+  ({ variant = "ghost", className, ...props }) => {
+    // TODO: compose this with forwardRef once @telegraph/composed-ref is published
+    const boxRef = React.useRef<HTMLDivElement>(null);
+
+    // Filter out the box props from the rest of the props
+    const filteredProps = Object.keys(props).reduce(
```

Comment:
I don’t think this is strictly necessary, but given how stable these props are is it worth memoizing this? Thinking about how often we’re going to be using these components and how many will exist in the tree it might be worth it?